py/server/tests/test_pyfunc_param_null.py

```diff
@@ -0,0 +1,170 @@
+    "long": "NULL_LONG",
+    "short": "NULL_SHORT",
+    "char": "NULL_CHAR",
+    "byte": "NULL_BYTE",
+}
+
+_J_TYPE_NP_DTYPE_MAP = {
+    "double": "np.float64",
+    "float": "np.float32",
+    "int": "np.int32",
+    "long": "np.int64",
+    "short": "np.int16",
+    "char": "np.uint16",
+    "byte": "np.int8",
+}
+
+
+class NullConvDecoratorTest(BaseTestCase):
+    def test_dh_null_to_nan_array(self):
+        def custom_func(col1, col2):
+            return np.mean(col2)
+
+        @dh_null_to_nan
+        def custom_func2(col1, col2):
+            return np.nanmean(col2)
+
+        col1_formula = "Col1 = i % 10"
+        for data_type, null_name in _J_TYPE_NULL_MAP.items():
+            col2_formula = f"Col2 = i % 3 == 0? {null_name} : ({data_type})i"
+            with self.subTest(data_type):
+                tbl = empty_table(100).update([col1_formula, col2_formula]).group_by("Col1")
+
+                exec(f"from deephaven.constants import {null_name}", globals())
+                custom_func3_str = f"lambda col1, col2: float(np.mean([c for c in col2 if c != {null_name}]))"
+
+                custom_func3 = eval(custom_func3_str, globals(), locals())
+
+                res = tbl.update("Col3 = custom_func(Col1, Col2)")
+                res2 = tbl.update("Col3 = custom_func2(Col1, Col2)")
+                res3 = tbl.update("Col3 = custom_func3(Col1, Col2)")
+                if data_type == "double" or data_type == "float":
+                    self.assertIn("-inf", res.to_string().lower())
+                    self.assertNotIn("-inf", res2.to_string().lower())
+                    self.assertNotIn("-inf", res3.to_string().lower())
+                else:
+                    self.assert_table_equals(res2, res3)
+
+                with self.assertRaises(AssertionError):
+                    self.assert_table_equals(res, res2)
+
+                exec(f"del {null_name}", globals())
+
+    def test_dh_null_to_nan_scalar(self):
+        col1_formula = "Col1 = i % 10"
+        for data_type, null_name in _J_TYPE_NULL_MAP.items():
+            col2_formula = f"Col2 = i % 3 == 0? {null_name} : ({data_type})i"
+            with self.subTest(data_type):
+                np_type = _J_TYPE_NP_DTYPE_MAP[data_type]
+                func = f"""
+@dh_null_to_nan
+def custom_func4(col: {np_type}) -> bool:
+    if np.isnan(col):
+        return True
+    else:
+        return False
+"""
+                exec(func, globals())
+                with self.subTest(data_type):
+                    tbl = empty_table(100).update([col1_formula, col2_formula])
+                    res = tbl.update("Col4 = custom_func4(Col2)")
+                    self.assertEqual(4, res.to_string().count("true"))
+
+                @dh_null_to_nan
+                @dh_vectorize
+                def custom_func5(col) -> bool:
+                    if np.isnan(col):
+                        return True
+                    else:
+                        return False
```

Comment:
It looks like scalars are being tested with (1) type hints and (2) with vectorize but not without type hints and without vectorize.  Seems like a missing test.